auacm/app/modules/submission_manager/judge.py

```diff
@@ -42,182 +46,169 @@
 
 
 def allowed_filetype(filename):
+    """Check to see if the filename is an allowed type or not"""
     return ('.' in filename and
             filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS)
 
+class Judge:
+    """A container for a judging instance. It contains the submission, the
+    source code that goes along with the submission, and the time limit of the
+    problem.
+    """
 
-def directory_for_submission(submission):
-    return os.path.join(
-        app.config['DATA_FOLDER'], 'submits', str(submission.job))
+    def __init__(self, submission, uploaded_file, time_limit):
+        """Create a new Judgement instance.
 
+        :param submission: the submission to be judged
+        :param uploaded_file: the source code file
+        :param time_limit: the time limit for execution
+        """
+        self.submission = submission
+        self.uploaded_file = uploaded_file
+        self.directory_for_submission = os.path.join(
+            app.config['DATA_FOLDER'], 'submits', str(self.submission.job))
+        self.directory_for_problem = os.path.join(
+            app.config['DATA_FOLDER'], 'problems', str(self.submission.pid))
+        self.time_limit = time_limit
 
-def directory_for_problem(submission):
-    return os.path.join(app.config['DATA_FOLDER'], 'problems', str(submission.pid))
 
+    def run_threaded(self):
+        """Runs the Judgement on a new thread"""
+        thread = threading.Thread(target=self.run, args=(self))
```

Comment:
The `args=(self)` is unnecessary and actually breaks. Didn't catch it in the unit tests because `run_threaded()` never gets used (maybe it should get added).